extensions/rocksdb-repos/ProvenanceRepository.h

```diff
@@ -72,7 +72,7 @@ class ProvenanceRepository : public core::Repository, public std::enable_shared_
     if (running_)
       return;
     running_ = true;
-    thread_ = std::thread(&ProvenanceRepository::run, shared_from_this());
+    thread_ = std::thread(&ProvenanceRepository::run, this);
```

Comment:
I can confirm that this fixes the ProvenanceRepository crash. Are the tests going to be in this pull request, or in a separate one?

Reply:
They are supposed to be in this one. Unless it is urgent to merge, then it's free to merge and then I'll open another PR.

Reply:
I don't think it's urgent.  I'll wait for the tests before approving, then.

Reply:
I think the mac failure comes from the main branch. Something caused that test to fail about half the time on mac.